### ISSUE TITLE
There is no store type corresponding to the conceptual side type 'DateTimeOffset' of primitive type 'DateTimeOffset'

### ISSUE THREAD

**Issue by raxitr** [IdentityServer/IdentityServer3.EntityFramework#60]
I am getting this error while adding EF code first migration on ClientConfigurationDbContext

I am using SQL server 2012 and EF 6. Not sure what is the issue, I have double check the connection string in App.Config which points to correct database.

here is what I have tried
1) extended class from ClientConfigurationDbContext so that Code First migration detects it.
2) run enable-migrations
3) got the error. 

Looks like it is from ```public virtual DateTimeOffset? Expiration { get; set; }``` in ```ClientSecret```. It should not but note sure why?

**Comment by brockallen** [comment]
Are you on 2.0.0?

**Comment by raxitr** [comment]
Do you mean IdentityServer? then No, its 1.4.0

**Comment by brockallen** [comment]
So whatever database you're using must not support DateTiimeOffset. This thread had a similar issue: https://github.com/IdentityServer/IdentityServer3.EntityFramework/issues/58

**Comment by raxitr** [comment]
SQL 2012 does support DateTimOffset so not sure what else it could be.

**Comment by brockallen** [comment]
Then last thing just might be an out of date schema? Have you been adding migrations? The same here has them:

https://github.com/IdentityServer/IdentityServer3.Samples/tree/master/source/EntityFramework

**Comment by raxitr** [comment]
No, I have not added any of the migrations since I am doing it for the first time. I believe it will create entire schema from scratch? Should take all these migrations in the sample and run them?

**Comment by brockallen** [comment]
If you're starting with the latest version then you don't need to apply the migrations (though once you get this working you should use migrations).

**Comment by raxitr** [comment]
I have started with latest version.

**Comment by brockallen** [comment]
Actually, the latest is 1.6.2: https://www.nuget.org/packages/Thinktecture.IdentityServer3
And EF is: 1.2.1: https://www.nuget.org/packages/Thinktecture.IdentityServer3.EntityFramework/

**Comment by raxitr** [comment]
More details,

EF is 1.2.1
Idserver is 1.4.2




```
Enable-Migrations -ContextTypeName Account.IdentityServer.Data.Library.ClientDbContext
```

```
public class ClientDbContext : ClientConfigurationDbContext 
    {
        public ClientDbContext():
            this(Constants.ConnectionName)
        {}
        
        public ClientDbContext(string connectionString):
            base(connectionString)
        {}
    }

```
```

Checking if the context targets an existing database...
System.NotSupportedException: There is no store type corresponding to the conceptual side type 'DateTimeOffset' of primitive type 'DateTimeOffset'.
   at System.Data.Entity.SqlServer.SqlProviderManifest.GetStorePrimitiveTypeIfPostSql9(String storeTypeName, String nameForException, PrimitiveTypeKind primitiveTypeKind)
   at System.Data.Entity.SqlServer.SqlProviderManifest.GetStoreType(TypeUsage edmType)
   at System.Data.Entity.ModelConfiguration.Edm.Services.StructuralTypeMappingGenerator.MapTableColumn(EdmProperty property, String columnName, Boolean isInstancePropertyOnDerivedType)
   at System.Data.Entity.ModelConfiguration.Edm.Services.PropertyMappingGenerator.Generate(EntityType entityType, IEnumerable`1 properties, EntitySetMapping entitySetMapping, MappingFragment entityTypeMappingFragment, IList`1 propertyPath, Boolean createNewColumn)
   at System.Data.Entity.ModelConfiguration.Edm.Services.TableMappingGenerator.Generate(EntityType entityType, DbDatabaseMapping databaseMapping)
   at System.Data.Entity.ModelConfiguration.Edm.Services.DatabaseMappingGenerator.GenerateEntityTypes(DbDatabaseMapping databaseMapping)
   at System.Data.Entity.ModelConfiguration.Edm.Services.DatabaseMappingGenerator.Generate(EdmModel conceptualModel)
   at System.Data.Entity.DbModelBuilder.Build(DbProviderManifest providerManifest, DbProviderInfo providerInfo)
   at System.Data.Entity.DbModelBuilder.Build(DbConnection providerConnection)
   at System.Data.Entity.Internal.LazyInternalContext.CreateModel(LazyInternalContext internalContext)
   at System.Data.Entity.Internal.RetryLazy`2.GetValue(TInput input)
   at System.Data.Entity.Internal.LazyInternalContext.InitializeContext()
   at System.Data.Entity.Internal.LazyInternalContext.get_ModelBeingInitialized()
   at System.Data.Entity.Infrastructure.EdmxWriter.WriteEdmx(DbContext context, XmlWriter writer)
   at System.Data.Entity.Utilities.DbContextExtensions.<>c__DisplayClass1.<GetModel>b__0(XmlWriter w)
   at System.Data.Entity.Utilities.DbContextExtensions.GetModel(Action`1 writeXml)
   at System.Data.Entity.Utilities.DbContextExtensions.GetModel(DbContext context)
   at System.Data.Entity.Migrations.DbMigrator..ctor(DbMigrationsConfiguration configuration, DbContext usersContext, DatabaseExistenceState existenceState, Boolean calledByCreateDatabase)
   at System.Data.Entity.Migrations.DbMigrator..ctor(DbMigrationsConfiguration configuration)
   at System.Data.Entity.Migrations.Design.MigrationScaffolder..ctor(DbMigrationsConfiguration migrationsConfiguration)
   at System.Data.Entity.Migrations.Design.ToolingFacade.ScaffoldRunner.Run()
   at System.AppDomain.DoCallBack(CrossAppDomainDelegate callBackDelegate)
   at System.AppDomain.DoCallBack(CrossAppDomainDelegate callBackDelegate)
   at System.Data.Entity.Migrations.Design.ToolingFacade.Run(BaseRunner runner)
   at System.Data.Entity.Migrations.Design.ToolingFacade.ScaffoldInitialCreate(String language, String rootNamespace)
   at System.Data.Entity.Migrations.EnableMigrationsCommand.<>c__DisplayClass2.<.ctor>b__0()
   at System.Data.Entity.Migrations.MigrationsDomainCommand.Execute(Action command)
There is no store type corresponding to the conceptual side type 'DateTimeOffset' of primitive type 'DateTimeOffset'.

```

**Comment by brockallen** [comment]
Right -- this tells me you have mismatched assemblies. Upgrade the IdSvr assembly.

**Comment by raxitr** [comment]
Updated but same error

**Comment by brockallen** [comment]
Hmm, ok... I'll try to repro when I can. Do the samples not work for you?

**Comment by raxitr** [comment]
I am  getting build error in sample https://github.com/IdentityServer/IdentityServer3.Samples/tree/master/source/EntityFramework

NuGet Package restore failed for project EntityFramework: Unable to find version '1.2.1-build00009' of package 'Thinktecture.IdentityServer3.EntityFramework'

**Comment by brockallen** [comment]
Update to the latest on the myget feed.

**Comment by brockallen** [comment]
Any update (hopefully success)?

**Comment by raxitr** [comment]
Yes the errors have gone away with the update from MyGet. I still need to run migrations.

**Comment by raxitr** [comment]
All sorted. I have followed instructions given here https://identityserver.github.io/Documentation/docs/ef/migrations.html 

Not sure what was the issue though,  
1) my startup project did not have the connection string IdSvr3Config
2) I have installed IdServer EF in class lib project and hence kept the connection string in app.config.
3) According to documentation I have enabled the migration which gave me the error of connection string not being present.
4) then I have added connection string in Startup project (i.e. Identity server project's web.config) 
5) tried enabling migration as per the documentation
6) All worked fine.

EF: 1.2.1
Identity Server : 1.6.2
